awx/main/isolated/manager.py

```diff
@@ -40,6 +42,51 @@ def __init__(self, cancelled_callback=None, check_callback=None):
         self.idle_timeout = max(60, 2 * settings.AWX_ISOLATED_CONNECTION_TIMEOUT)
         self.started_at = None
         self.captured_command_artifact = False
+        self.instance = None
+
+    def build_inventory(self, hosts):
+        if self.instance and self.instance.is_containerized:
+            inventory = { 'all': { 'hosts': {} } }
+            for host in hosts:
+                inventory['all']['hosts'][host] = self.kubectl_or_oc_vars
+        else:
+            inventory = '\n'.join([
+                '{} ansible_ssh_user={}'.format(host, settings.AWX_ISOLATED_USERNAME)
+                for host in hosts
+            ])
+
+        return inventory
+
+    @property
```

Comment:
I don't _love_ that this is a @property . I like to make things a function contain any side effects. In this case, a file is created on disk. Multiple calls to this will result in the ssl ca cert getting written to disk multiple times.

Reply:
My opinion on this is, from a perspective of application state, this function does not have any side-effects. The fact that it generates a tempfile is irrelevant to the interface and will always return the same output. It does not modify *other* properties of the object, which is the kind of thing I do think we should avoid in a property. That being said, I see your point and I'm on the fence about whether to care or not, so I'll go ahead and give this one to you.